src/ts/contentScripts/duplicate-block-content/index.tsx

```diff
@@ -1,13 +1,32 @@
 import { Feature } from '../../utils/settings'
-import { Roam } from '../../utils/roam';
+import { Roam, RoamNode, Selection } from '../../utils/roam';
 
 export const config: Feature = {
     id: 'duplicate',
     name: 'Duplicate',
     shortcuts: [
         {
             id: 'dupShortcut', label: 'Shortcut for duplication', initValue: '',
-            onPress: () => Roam.duplicateBlock()
+            onPress: () => duplicate()
         },
     ]
+}
+
+const duplicate = () => {
+    const node = Roam.getActiveRoamNode();
+    const selectedText = node?.selectedText();
+
+    if (node && selectedText) {
+        const selectionLength = selectedText.length;
+        const selectionEnd = node.selection.end
+        const newText = node.text.substring(0, selectionEnd)
+            + selectedText
+            + node.text.substring(selectionEnd);
+
+        Roam.applyToCurrent((() => {
```

Comment:
the whole point of "apply to current" is that you know that if you get inside that function - roam text node exists. so you don't need to retrieve it manually/do checks on whether it's valid/etc

Reply:
forgot there's save()